Fix #6226: Added closestWidgetVar EL function

jepsar opened the pull request:
Fixes #6226

jepsar committed:
Added closestWidgetVar EL function

jepsar commented on src/main/java/org/primefaces/expression/SearchExpressionUtils.java:
```diff
@@ -85,10 +87,30 @@ public static String resolveWidgetVar(String expression, UIComponent component)
     }
+
+    // used by p:closestWidgetVar
+    public static String closestWidgetVar() {
```
I wasn't sure where to put these methods... is it okay to have them here?

melloware reviewed (approved):
Looks good to me but will let Thomas review

tandraschko commented:
i dont like your change that you automatically add PF('') now
this breaks backward compatibility and doesnt allow to just resolve the widgetVar anymore, without having PF('') around

jepsar commented:
> i dont like your change that you automatically add PF('') now
> this breaks backward compatibility and doesnt allow to just resolve the widgetVar anymore, without having PF('') around

I didn't change that, I just moved it to a method to DRY things up and fixed the documentation.

https://github.com/primefaces/primefaces/blob/de9a16e875e66c7233aa30fb714b3be7409c2c02/src/main/java/org/primefaces/expression/SearchExpressionUtils.java#L88

.. but I agree it's a bit odd. I was surprised it was working like that. Shall I remove it?

tandraschko commented:
oha, i see
please, it should just return the varName, not PF('')

tandraschko commented:
and IMO we should not create 2 methods for it with different names, we had it in the past and its dirty

i think we have 2 solutions:
- test overloading with all current containers (it didnt work in the past)
or
- always pass the UIComponent like we did for resolveClientId. So you always need: #{p:closestWidgetVar(component)}

#{component} like like UIComponet.getCurrentComponent

jepsar committed:
Removed PF('') wrapper

jepsar committed:
Removed closestWidgetVarFor

jepsar commented:
@tandraschko done

tandraschko commented:
Please also adjust the migration guide that people must add PF('') now

jepsar commented:
> Please also adjust the migration guide that people must add PF('') now

Done as well
